backend/djangoindia/api/views/communication.py

```diff
@@ -31,6 +31,30 @@ def post(self, request, *args, **kwargs):
             return Response(
                 {"message": str(e)}, status=status.HTTP_500_INTERNAL_SERVER_ERROR
             )
+class UnsubscribeAPIView(generics.GenericAPIView):
+    def delete(self, request, *args, **kwargs):
+        try:
+            email = request.data.get("email")
+            if not email:
+                return Response(
+                    {"message": "Email address is required."},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+            subscriber = Subscriber.objects.filter(email=email).first()
+            if not subscriber:
+                return Response(
+                    {"message": "No subscription found for this email."},
+                    status=status.HTTP_404_NOT_FOUND,
+                )
```

Comment:
nit: both can be handled in a single condition maybe. please check and update accordingly

Reply:
ok will update it

Reply:
You're right! Both conditions can be combined. I'll refactor the logic accordingly